### DISCUSSION TITLE
FFTW in a Firedrake dockerfile

### DISCUSSION THREAD

**Discussion by colinjcotter** [firedrakeproject/firedrake#2330] [Firedrake support] [answered]
Hi folks,

I use the Firedrake docker file in the automated test system for https://github.com/colinjcotter/asQ via GitHub Actions.
We need to build Firedrake by passing PETSC_CONFIGURE_OPTIONS="--download-fftw=1" to the install script, because we use FFTW and it needs to be 

Please can you provide an FFTW version of the docker file? Or if that's not a good idea, please can you tell me how I can provide one?

all the best
--cjc

**Comment by dham** [comment, marked as answer]
I can't see any reason not to have FFTW in the standard Docker containers. Put in a PR adding to the install line in `docker/Dockerfile.vanilla`

**Reply by wence-** [reply]
(Done in https://github.com/firedrakeproject/firedrake/pull/2331)

**Comment by colinjcotter** [comment]
I have included the following lines in my test.yml file that specifies the GitHub actions:
```
          export FFTW_INCLUDE_DIR=/home/firedrake/src/petsc/default/include
          export FFTW_LIBRARY_DIR=/home/firedrake/src/petsc/default/lib
          python -m pip install mpi4py-fft 
```
The mpi4py-fft installer uses those variables to link against your specified FFTW build. However, the GitHub Actions install says:

```
AssertionError: No FFTW libraries found in ['/home/firedrake/firedrake/lib64', '/home/firedrake/firedrake/lib', '/usr/lib64', '/usr/lib/x86_64-linux-gnu', '/usr/lib']
```
This suggests that these variables are not getting passed to pip install. What is the right way to do this?

**Reply by wence-** [reply]
does the docker image actually have fftw yet?

**Reply by wence-** [reply]
It should be `/home/firedrake/firedrake/src/petsc/default/include` etc...
